source/Octopus.Cli/Commands/InstallAutoCompleteCommand.cs

```diff
@@ -0,0 +1,181 @@
+                {
+                    commandOutputProvider.Information("Looks like this is already installed. Bailing out.");
+                    return;
+                }
+
+                commandOutputProvider.Information($"Updating profile at {profilePath}");
+                tempOutput.AppendLine(profileText);
+            }
+            else
+            {
+                commandOutputProvider.Information($"Creating profile at {profilePath}");
+            }
+
+            tempOutput.AppendLine(UserProfileHelper.AllShellsPrefix);
+            tempOutput.AppendLine(scriptToInject);
+            tempOutput.AppendLine(UserProfileHelper.AllShellsSuffix);
+
+            if (DryRun)
+            {
+                commandOutputProvider.Warning("Preview of script changes: ");
+                commandOutputProvider.Information(tempOutput.ToString());
+                commandOutputProvider.Warning("Preview of script changes finished. ");
+            }
+            else
+            {
+                fileSystem.OverwriteFile(profilePath, tempOutput.ToString());
+                commandOutputProvider.Warning("All Done! Please reload your shell or dot source your profile to get started! Use the <tab> key to autocomplete.");
+            }
+        }
+        
+        private void InstallForBash()
+        {
+            Install(
+                UserProfileHelper.BashProfile, 
+                UserProfileHelper.BashProfileScript);
+        }
+
+        private void InstallForZsh()
+        {
+            Install(
+                UserProfileHelper.ZshProfile, 
+                UserProfileHelper.ZshProfileScript);
+        }
+
+        private void InstallForPowershell()
+        {
+            var profilePath = UserProfileHelper.GetPwshProfileForOperatingSystem();
+            Install(profilePath, UserProfileHelper.PwshProfileScript);
+        }
+    }
+    public static class UserProfileHelper
+    {
+        public static readonly string EnvHome = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "HOMEPATH" : "HOME";
+        public static readonly string HomeDirectory = System.Environment.GetEnvironmentVariable(EnvHome);
+        public static readonly string ZshProfile = $"{HomeDirectory}/.zshrc";
+        public const string ZshProfileScript = 
+@"_octo_zsh_complete()
+{
+    local completions=(""$(octo complete $words)"")
+    reply=( ""${(ps:\n:)completions}"" )
+}
+compctl -K _octo_zsh_complete octo";
+
+        public static readonly string PwshProfileLocationWindows = $"{HomeDirectory}\\Documents\\WindowsPowerShell\\Microsoft.PowerShell_profile.ps1";
+        public static readonly string PwshProfileLocationNix =  $"{HomeDirectory}/.config/powershell/Microsoft.PowerShell_profile.ps1";
+        public const string PwshProfileScript =
+@"Register-ArgumentCompleter -Native -CommandName octo -ScriptBlock {
+    param($wordToComplete, $commandAst, $cursorPosition)
+    $parms = ($commandAst.ToString()).Replace('octo ','')
```

Comment:
what if they type `octo.exe`?

Reply:
I can make several registrations to cover that - thanks!

Reply:
Powershell handles this actually, the issue has to do with `octo` not being on the PATH. I'm looking at fully resolving this path to address that issue

Reply:
Trying to handle pathless installations is a right pain.

Microsoft don't even bother: https://github.com/dotnet/cli/blob/master/Documentation/general/tab-completion.md?WT.mc_id=-blog-scottha#how-to-enable-it

```
These scripts assume that dotnet v2.0 is on your path.
```

Given that the majority of installations will automatically install the binary on people's PATH I think I'll do the same. We can just document it well enough such that people motivated enough can modify these scripts themselves if they are manually installing to non-standard locations.